react/index.d.ts

```diff
@@ -37,7 +37,7 @@ declare namespace React {
 
     type CElement<P, T extends Component<P, ComponentState>> = ComponentElement<P, T>;
     interface ComponentElement<P, T extends Component<P, ComponentState>> extends ReactElement<P> {
-        type: ComponentClass<P>;
+        type: ComponentClass<P> | ComponentClass<any>;
```

Comment:
Yea this is kind of sensitive, but that's what code reviews are for, I'm kind of unsure how to best fix this.

![screenshot from 2016-10-30 15-57-08](https://cloud.githubusercontent.com/assets/6056334/19837435/a5940d1e-9eb9-11e6-9423-30bf919488b4.png)